muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/actions/main-function/users/index.ts

```diff
@@ -34,34 +52,29 @@ export interface LoadStudyprogrammesTriggerType {
 }
 
 export interface LoadUsersTriggerType {
-  (): AnyActionType
+  (q?: string): AnyActionType
 }
 
-const selectStudentEndPoint:  ApplyStudentTriggerType = function studentEndPoint (data) {
-  if(data.mode  == "CREATE" ) {
-    return mApi().user.students.create(data.student);
-  } else {
-    return mApi().user.students.put(data.student);
-  }
-}
+let createStudent: CreateStudentTriggerType = function createStudent(data) {
+  return async (dispatch: (arg: AnyActionType) => any, getState: () => StateType) => {
+    try {
+      await promisify(mApi().user.students.create(data.student), 'callback')().then(() => {
 
-let applyStudent: ApplyStudentTriggerType = function applyStudent (data) {
+        mApi().organizationUserManagement.students.cacheClear();
 
-  return async (dispatch:(arg:AnyActionType)=>any, getState:()=>StateType)=>{
-    try {
-      await promisify(selectStudentEndPoint(data), 'callback')().then(() => {
-        promisify(mApi().user.students.read(), 'callback')()
-        .then((students:UserType) => {
+        setTimeout(async () => {
```

Comment:
Because create :  muikku --> pyramus ---> muikku --> and if it just refreshes, it's too fast and no changes in state. I made it 2 second just to be sure.